### PR TITLE
Create gui

### PR DESCRIPTION
**Pull request by ChiliEater**

### CODE REVIEW

File: src/scenes/ui/BattleUI.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    public override void _Ready()
+    {
+        
+    }
+
+    public void onTowerSelected(TowerButton towerButton) {
+        if (towerButton != null) {
+            currentTower = towerButton.TowerType;
+        }
```
Comment (manuth) [inline src/scenes/ui/BattleUI.cs:16]: You might want to use null conditionals here:

```cs
currentTower = towerButton?.TowerType;
```

File: src/scenes/ui/BattleUI.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        foreach (Node node in GetChildren())
+        {
+            if (node is TowerButton) {
+                TowerButton tower = (TowerButton)node;
```
Comment (manuth) [inline src/scenes/ui/BattleUI.cs:20]: You might want to simplify this statement:

```cs
if (node is TowerButton tower) {
    // Your code here
}
```

File: src/scenes/ui/BattleUI.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+                tower.Active = tower == towerButton ? true : false;
```
Comment (manuth) [inline src/scenes/ui/BattleUI.cs:21]: The `? true : false` part is not necrssary. Consider to remove it

**Review by manuth — CHANGES_REQUESTED** [review]
Looks fine to me overall!
Great job 😄

I have a few changes to ask for before ready for approval.

File: src/scenes/ui/BattleUI.cs (hunk 3 above)
Comment (ChiliEater) [reply src/scenes/ui/BattleUI.cs:21]: oh my god

**Review by manuth — APPROVED** [review]
Looking good - great job 😄

**Pull request merged**